src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -590,9 +590,7 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
                     $object,
                     $cacheKey.':'.$propertyName,
                     $propertyMetadata,
-                    $propertyPath
-                        ? $propertyPath.'.'.$propertyName
-                        : $propertyName,
+                    PropertyPath::append($propertyPath, $propertyName),
```

Comment:
Makes sense, it doesn't look like there is a unit test for RecursiveContextualValidator already though?